File: solutions/javascript/word-count/2/word-count.js
export default class Words {
    count(phrase) {
        return phrase.trim().toLowerCase().split(/\s+/).reduce((counts, word) => {
            if (counts.hasOwnProperty(word)) { counts[word]++; }
            else { counts[word] = 1; }
            return counts;
        }, {});
    }
}

File: solutions/clojure/word-count/2/src/word_count.clj
(ns word-count
  (:require [clojure.string :as str]))

(def word-count
  (comp
    frequencies ; create a hashmap of frequencies
    #(str/split % #"\W+") ; from words split by multiple non-[a-zA-Z0-9_] characters
    str/lower-case) ; from the input converted to lower-case
)